File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/JAVA_TEMA/tema_bun/Persistence/src/main/java/com/example/tema_bun/ProbaRepository.java
package com.example.tema_bun;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;


public class ProbaRepository implements Repository<String, Proba>{

    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public ProbaRepository(Properties props) {
        logger.info("Initializing ProbaRepository with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }

    @Override
    public void adauga(Proba elem) {
        //to do
        logger.traceEntry("saving task {} ",elem);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into Probe(id) values (?)")){
            //preStmt.setInt(1,elem.getId());
            preStmt.setString(1,elem.getId());
            int result=preStmt.executeUpdate();
            logger.trace("Saved {} instances",result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB "+ex);
        }
        logger.traceExit();
    }

    @Override
    public void update(String id, Proba elem) {
        //to do
//        logger.traceEntry("updating elem {}",elem);
//        Connection con = dbUtils.getConnection();
//        try(PreparedStatement preStmt = con.prepareStatement("update Probe set nume=? where id=?")){
//            preStmt.setString(1,elem.getNume());
//            preStmt.setInt(2, id);
//
//            int result=preStmt.executeUpdate();
//            logger.trace("Updated {} instances",result);
//        } catch (SQLException e) {
//            logger.error(e);
//            System.err.println("DB Error"+e);
//        }
    }

    @Override
    public List<Proba> getAll() {
        //to do
        logger.traceEntry("gettig find all");
        Connection con=dbUtils.getConnection();
        List<Proba> probe = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from Probe"))
        {
            try(ResultSet result = preStmt.executeQuery())
            {
                while(result.next())
                {
                    String id = result.getString("id");
                    //String nume = result.getString("nume");
                    Proba proba = new Proba(id);
                    //proba.setId(id);
                    probe.add(proba);
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB"+e);
        }
        logger.traceExit(probe);
        return probe;
    }

    public void sterge(Proba elem){
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("delete from Probe where id=?")){
            preStmt.setString(1, elem.getId());

            preStmt.executeUpdate();
        }
        catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB " + e);
        }
    }

    public Proba cautaId(String idCautat){
        //to do
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        Proba proba;
        try(PreparedStatement preStmt = con.prepareStatement("select * from Probe where id=?")){
            preStmt.setString(1,idCautat);
            try(ResultSet result=preStmt.executeQuery()){
                if (result.next()) {
                    String id = result.getString("id");
                    //String nume = result.getString("nume");

                    proba = new Proba(id);
                    return proba;
                }
            }
        }
        catch(SQLException e){
            logger.error(e);
            System.err.println("Error DB " + e);
        }

        return null;
    }
}

File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/CSHARP_TEMA/tema_sapt3/Persistance/ProbaRepository.cs
using log4net;
using Model;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace persistance
{
    public class ProbaRepository : Repository<String, Proba>
    {
        private readonly ILog _logger = LogManager.GetLogger(nameof(ProbaRepository));
        public void adauga(Proba entity)
        {
            //_logger.Info("Adauga proba");
            //var connection = ConnectionUtils.CreateConnection();
            //var command = connection.CreateCommand();
            //command.CommandText = "insert into Probe(nume) values (@nume)";
            //command.Parameters.AddWithValue("@nume", entity.nume);

            //try
            //{
            //    command.ExecuteNonQuery();
            //}
            //catch (Exception ex)
            //{

            //}

        }

        public List<Proba> getAll()
        {
            _logger.Info("get all probe");
            var list = new List<Proba>();
            var connection = ConnectionUtils.CreateConnection();
            var command = connection.CreateCommand();
            command.CommandText = "select * from Probe";
            var reader = command.ExecuteReader();
            while (reader.Read())
            {
                var id = reader.GetString(0);
                //var nume = reader.GetString(1);
                var proba = new Proba()
                {
                    id = id,
                    //nume = nume
                };
                list.Add(proba);
            }
            return list;
        }

        public Proba cautaId(String id)
        {
            var list = getAll();
            foreach (var el in list)
            {
                if (el.id == id)
                {
                    return el;
                }
            }

            return null;
        }

        public void sterge(Proba entity)
        {
            _logger.Info("sterge" +
                " proba");
            var connection = ConnectionUtils.CreateConnection();
            var command = connection.CreateCommand();
            command.CommandText = "delete from Probe where id=@id";
            command.Parameters.Add("@id", DbType.String);
            command.Parameters["@id"].Value = entity.id;

            try
            {
                command.ExecuteNonQuery();
            }
            catch (Exception ex)
            {

            }
        }

        public void update(Proba entitate, Proba nouaEntitate)
        {
            //_logger.Info("update proba");
            //var connection = ConnectionUtils.CreateConnection();
            //var command = connection.CreateCommand();
            //command.CommandText = "update Probe set nume=@nume where id=@id";
            //command.Parameters.Add("@nume", DbType.String);
            //command.Parameters["@nume"].Value = nouaEntitate.nume;
            //command.Parameters.Add("@id", DbType.String);
            //command.Parameters["@id"].Value = entitate.id;
            //try
            //{
            //    command.ExecuteNonQuery();
            //}
            //catch (Exception ex)
            //{

            //}
        }
    }
}
